atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraTimestampsEteTest.java

```diff
@@ -87,12 +87,42 @@ public void timestampsForRangeTombstonesAreCurrentInThoroughTables() throws IOEx
         assertDroppableTombstoneRatioPositive(sstableMetadata);
     }
 
-    private void sweepUntilNoValueExistsAtTimestamp(long id, long timestamp) {
+    @Test
+    public void timestampsForSentinelsAndTombstonesAreCurrentInConservativeTables()
+            throws IOException, InterruptedException {
+        long firstWriteTimestamp = todoClient.addNamespacedTodoWithIdAndReturnTimestamp(ID, NAMESPACE, TODO);
+        todoClient.addNamespacedTodoWithIdAndReturnTimestamp(ID, NAMESPACE, TODO_2);
+        sweepUntilNoValueExistsAtTimestamp(ID, firstWriteTimestamp, Optional.of(NAMESPACE));
+
+        CassandraCommands.nodetoolFlush(CASSANDRA_CONTAINER_NAME);
+
+        List<String> ssTables = CassandraCommands.nodetoolGetSSTables(
+                CASSANDRA_CONTAINER_NAME,
+                "atlasete",
+                TodoSchema.namespacedTodoTable(),
+                ValueType.STRING.convertFromJava(NAMESPACE));
+        String sstableMetadata = CassandraCommands.ssTableMetadata(
+                CASSANDRA_CONTAINER_NAME, Iterables.getOnlyElement(ssTables));
+
+        // The table should contain the first value, and timestamps thereafter should be higher.
+        // Failure here means either that the range tombstone was written at a lower Cassandra timestamp than the
+        // original value (unlikely since we confirmed no value existed at the first write timestamp), or
+        // that the sweep sentinel was written at a lower Cassandra timestamp than the original value (and hence not
+        // a fresh timestamp).
+        assertMinimumTimestampIsAtLeast(sstableMetadata, firstWriteTimestamp);
+
+        // Failure here means that the range tombstone isn't droppable, which would be weird
+        assertDroppableTombstoneRatioPositive(sstableMetadata);
+    }
+
+    private void sweepUntilNoValueExistsAtTimestamp(long id, long timestamp, Optional<String> namespace) {
```

Comment:
The optional parameter is slightly hacky here. I guess you're trying to avoid duplicating the Awaitility logic, but this could be done without an optional parameter by having a helper method that takes a boolean, e.g.
```
private void sweepUntilNoValueExistsForNamespaceAtTimestamp(long id, String namespace, long timestamp) {
    sweepUntilConditionSatisfied(
        () -> todoClient.namespacedTodoDoesNotExistBeforeTimestamp(id, timestamp, namespace));
}
```

Reply:
Yep, having a BooleanSupplier condition works well! Changed to use one of these